pandoc_mermaid_filter.py

```diff
@@ -38,13 +40,15 @@ def mermaid(key, value, format_, _):
                 if os.path.isfile('.puppeteer.json'):
                     cmd.extend(["-p", ".puppeteer.json"])
 
-                subprocess.check_call(cmd)
+                subprocess.check_call(cmd, timeout=120)
```

Comment:
What if it's going to take longer than those 2 minutes?

Reply:
I agree - I'm less concerned about the time here than the concept of a timeout. I had a bunch of documents hang a CI process (which gave up after an hour anyway) because the diagram generation from pandoc-mermaid-filter encountered a blocking issue.

```
toJSONFilter(mermaid)
  File "/root/.cache/pypoetry/virtualenvs/proj-py3.6/lib/python3.6/site-packages/pandocfilters.py", line 132, in toJSONFilter
    toJSONFilters([action])
  File "/root/.cache/pypoetry/virtualenvs/proj-py3.6/lib/python3.6/site-packages/pandocfilters.py", line 166, in toJSONFilters
    sys.stdout.write(applyJSONFilters(actions, source, format))
BlockingIOError: [Errno 11] write could not complete without blocking
```